### PR TITLE
Emit Windows events from libiothsm

### PR DESCRIPTION
**Pull request by arsing**
With this change, all libiothsm logs are also emitted as Windows events
in addition to being written to stdout, unless iotedged is running in
console mode.

This also changes the default log level of libiothsm to "information"
to match iotedged's setting.

### CODE REVIEW

File: edgelet/hsm-sys/azure-iot-hsm-c/src/hsm_log.c [hunk 1]
```diff
@@ -30,5 +66,30 @@ void log_msg(int level, const char* file, const char* function, int line, const
         vsnprintf(buffer, MAX_LOG_SIZE, fmt_str, args);
         printf("<%d>%s [%s] (%s:%s:%d) %s\r\n", syslog_levels[level], time_buf, levels[level], file, function, line, buffer);
         va_end (args);
+
+#if defined __WINDOWS__ || defined _WIN32 || defined _WIN64 || defined _Windows
+        if (event_log_handle != NULL) {
+            size_t log_length = strlen(file) + strlen(function) + strlen(buffer) + 50;
```
Comment (mrohera) [inline edgelet/hsm-sys/azure-iot-hsm-c/src/hsm_log.c]: Can you please name the magic number with rationale as to how you arrived at it?

Comment (darobs) [reply edgelet/hsm-sys/azure-iot-hsm-c/src/hsm_log.c]: I wish C had a better way of finding the expected length of a format string. that would be cool.

Comment (arsing) [reply edgelet/hsm-sys/azure-iot-hsm-c/src/hsm_log.c]: Done.

**Comment by arsing** [comment]
This broke the Windows ARM32 build because the linker can't find the two win32 functions `ReportEventA` and `RegisterEventSourceA`.

```
hsm_log.obj : error LNK2019: unresolved external symbol __imp_RegisterEventSourceA referenced in function log_init [D:\a\1\s\edgelet\target\thumbv7a-pc-windows-msvc\release\build\hsm-sys-ca0b1d4729451947\out\build\tests\hsm_client_tpm_ut\hsm_client_tpm_ut_exe.vcxproj]
hsm_log.obj : error LNK2019: unresolved external symbol __imp_ReportEventA referenced in function log_msg [D:\a\1\s\edgelet\target\thumbv7a-pc-windows-msvc\release\build\hsm-sys-ca0b1d4729451947\out\build\tests\hsm_client_tpm_ut\hsm_client_tpm_ut_exe.vcxproj]
D:\a\1\s\edgelet\target\thumbv7a-pc-windows-msvc\release\build\hsm-sys-ca0b1d4729451947\out\build\tests\hsm_client_tpm_ut\Release\hsm_client_tpm_ut_exe.exe : fatal error LNK1120: 2 unresolved externals [D:\a\1\s\edgelet\target\thumbv7a-pc-windows-msvc\release\build\hsm-sys-ca0b1d4729451947\out\build\tests\hsm_client_tpm_ut\hsm_client_tpm_ut_exe.vcxproj]
```

It's probably another case of amd64 implicitly linking the import lib whereas ARM32 requires it explicitly, in which case we need to add `advapi32.lib` just like [we added `shell32` in the original Windows ARM32 commit.](https://github.com/Azure/iotedge/commit/dc473b132c99f338bfd0511762dbdd44fc72e31b#diff-79b86faa1ac3bcf410e82f699806624fR189-R193)

I'll fix it later today.

**Pull request merged**